interface/main/messages/messages.php

```diff
@@ -380,105 +382,117 @@ class='form-horizontal'
+                                                    $patientname = $prow['lname'] . ", " . $prow['fname'];
+                                                }
+                                                if ($task == "addnew" || $result['pid']==0) {
+                                                    $cursor = "oe-cursor-add";
+                                                    $background = "oe-patient-background";
+                                                } elseif ($task == "edit") {
+                                                    $cursor = "oe-cursor-stop";
+                                                    $background = '';
+                                                }
+                                                ?>
+                                                <input type='text'  id='form_patient' name='form_patient' class='form-control <?php echo $cursor . " " .$background;?>' onclick="multi_sel_patient()" placeholder='<?php echo xla("Click to add patient"); ?>' value='<?php echo attr($patientname); ?>' readonly/>
+                                                <input type='hidden' class="form-control" name='reply_to' id='reply_to' value='<?php echo attr($reply_to); ?>'/>
+                                            </div>
+                                            <div class="col-xs-2">
+                                                <?php
+                                                if ($task=="addnew" || $result['pid']==0) {
+                                                    echo "<label class='control-label oe-empty-label' for='clear_patients'></label>";
+                                                    echo '<button type="button" id="clear_patients"  class="btn btn-default btn-undo pull-left flip" value="' . xla('Clear') .'">' . xlt("Clear") . '</button>';
+                                                } ?>
+                                            </div>
+                                        </div>
                                     </div>
                                 </div>
-                                <div class='col-xs-12 oe-margin-t-3'>
+                                <div class="row">
+                                    <div class="col-xs-12 col-sm-12 col-md-12 col-lg-10 oe-custom-line col-lg-offset-1">
+                                        <div class="row">
+                                            <?php if ($GLOBALS['messages_feature_date']) { ?>
+                                            <div class="col-xs-6 col-sm-2">
+                                                <label class="control-label" for="form_note_type"><?php echo xlt('Date'); ?>:</label>
+                                                <?php generate_form_field(array('data_type' => 4, 'field_id' => 'datetime', 'edit_options' => 'F'), empty($datetime) ? date('Y-m-d H:i') : $datetime) ?>
```

Comment:
Are you just doing a standard datetime input field here or something special?

Reply:
Future date and format time - Y-m-d H:i
```
 $('.datetimepicker').datetimepicker({
                <?php $datetimepicker_timepicker = true; ?>
                <?php $datetimepicker_showseconds = false; ?>
                <?php $datetimepicker_formatInput = true; ?>
                <?php require($GLOBALS['srcdir'] . '/js/xl/jquery-datetimepicker-2-5-4.js.php'); ?>
                ,minDate : 0 //only future
            })
```

Reply:
Very smart use of generate_form_field for a date/time entry. I never thought to do that :)
Note you'll need to output the proper date format when it is empty rather than date('Y-m-d H:i'), which can do with:
```
oeFormatDateTime(date('Y-m-d H:i'))
```
(note this is automatically done in generate_form_field() when the entry is not blank)

Reply:
Disregard my comment above. The date('Y-m-d H:i') is being converted by generate_form_field()